Binaries/DafnyRuntime.h

```diff
@@ -73,6 +73,31 @@ class IntegerRange {
    iterator end_;
 };
 
+// Workaround the fact that Apple's clang and g++ print nullptr as 0x0,
+// while Linux's g++ prints it as 0
```

Comment:
Nit-picking for the record: both `0` and `0x0` would be wrong anyway. :) The fact that Dafny null values are materialized as `nullptr` in C++ is pure implementation detail, and although Dafny doesn't currently specify enough about how arbitrary values are string-ified by `print` statements, it should be the same for each target language.

Reply:
Agreed that ideally all the backends should agree on print formatting.

Reply:
It's ideal if printing can be consistent everywhere and, more to the point, print things that make sense Dafny-wise. However, in the end, the `print` statement is for less demanding printing tasks, where the exact formatting probably doesn't matter. A demanding user would want to have more control over formatting, and would want to print not just to stdout. So, we may not always meet this ideal.